libs/ha/ha_common_libs_k8s.py

```diff
@@ -1312,9 +1318,27 @@ def simulate_disk_cvg_failure(self, node_obj, source, resource_status, resource_
                 config_dict["events"][f"{count}"]["resource_type"] = resource_type
                 config_dict["events"][f"{count}"]["source"] = source
                 config_dict["events"][f"{count}"]["resource_status"] = resource_status
-                config_dict["events"][f"{count}"]["node_id"] = node_id_list[n_cnt]
-                config_dict["events"][f"{count}"]["resource_id"] = resource_id_list[d_cnt]
-
+                if node_id and resource_id:
+                    # If user provide node_id and resource_id
+                    config_dict["events"][f"{count}"]["node_id"] = node_id
+                    config_dict["events"][f"{count}"]["resource_id"] = resource_id
+                elif node_id and resource_type == 'node':
+                    # If user provide node_id
+                    config_dict["events"][f"{count}"]["node_id"] = node_id
+                    config_dict["events"][f"{count}"]["resource_id"] = node_id
+                elif resource_id and resource_type == 'node':
+                    # If user provide resource_id
+                    config_dict["events"][f"{count}"]["node_id"] = resource_id
+                    config_dict["events"][f"{count}"]["resource_id"] = resource_id
+                else:
+                    # If user don't provide anything
+                    config_dict["events"][f"{count}"]["node_id"] = node_id_list[n_cnt]
+                    config_dict["events"][f"{count}"]["resource_id"] = resource_id_list[d_cnt]
+                if specific_info:
+                    config_dict["events"][f"{count}"]["specific_info"] = {}
+                    config_dict["events"][f"{count}"]["specific_info"]['generation_id'] = generation_id
```

Comment:
We are not sure what else we can pass in specific_info, we should get this dict directly from user instead of adding key/value in lib. If we add each and every parameter which can part of specific_info here, it will be messy. Better to get dict filled by user itself.
If user provides specific_info dict, assign it here elif None then empty dict should be passed.

Reply:
Suggested changes are addressed.